spec/jacaranda_spec.rb

```diff
@@ -11,6 +11,42 @@ def all_request_bodies
 end
 
 describe 'Jacaranda' do
+  describe '.remove_old_tables!' do
+    before(:each) do
+      create_table_statements.each { |statement| ScraperWiki.sqliteexecute(statement) }
+    end
+
+    context 'when the old tables are present' do
+      let(:create_table_statements) do
```

Comment:
It looks like these aren't actually testing what they appear to because `let` doesn't actually run unless/until the variable is called https://relishapp.com/rspec/rspec-core/docs/helper-methods/let-and-let So potentially both these tests pass because the tables were never created.

Should we be using `let!` here?

Reply:
The `let` will run, because the `before(:each)` is at the `describe` level, not the `context` level.

This is done so each context can have a context-specific database setup.

Reply:
To demonstrate this, there's [an explicit test](https://github.com/openaustralia/jacaranda/pull/20/files#diff-73ddc23149847f29ab48e2bdfcfb41e7R29) that the table exists before running the `remove_old_tables!` method.

Reply:
Nice 👍